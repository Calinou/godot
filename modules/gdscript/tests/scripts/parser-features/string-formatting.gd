func test():
	assert("hello %s" % "world" == "hello world")
	assert("hello %s" % true == "hello True")
	assert("hello %s" % false == "hello False")

	assert("hello %d" % 25 == "hello 25")
	assert("hello %d %d" % [25, 42] == "hello 25 42")
	# Pad with spaces.
	assert("hello %3d" % 25 == "hello  25")
	# Pad with zeroes.
	assert("hello %03d" % 25 == "hello 025")

	assert("hello %.02f" % 0.123456 == "hello 0.12")

	# Dynamic padding:
	# <https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_format_string.html#dynamic-padding>
	assert("hello %*.*f" % [7, 3, 0.123456] == "hello   0.123")
	assert("hello %0*.*f" % [7, 3, 0.123456] == "hello 000.123")
