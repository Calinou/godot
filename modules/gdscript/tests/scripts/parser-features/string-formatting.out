GDTEST_OK
>> WARNING
>> Line: 2
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 3
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 4
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 6
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 9
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 11
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 13
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
